src/webgl/uniforms.js

```diff
@@ -135,53 +118,37 @@ const UNIFORM_SETTERS = {
 };
 
 // Pre-allocated typed arrays for temporary conversion
-const FLOAT_ARRAY = [2, 3, 4, 6, 8, 9, 12, 16].reduce((arrays, length) => {
-  arrays[length] = new Float32Array(length);
-  return arrays;
-}, {});
-const INT_ARRAY = {
-  2: new Int32Array(2),
-  3: new Int32Array(3),
-  4: new Int32Array(4)
-};
-const UINT_ARRAY = {
-  2: new Uint32Array(2),
-  3: new Uint32Array(3),
-  4: new Uint32Array(4)
-};
+const FLOAT_ARRAY = {};
```

Comment:
Prefer if we only keep a single array, the high water mark.

Reply:
Does that mean we will upload a bigger chunk of data to GPU in every uniform setter?

Reply:
> Does that mean we will upload a bigger chunk of data to GPU in every uniform setter?

I see your point. The shader compiler should have parsed the uniforms (including the array specification) and assigned them to the minimal number of `vec4`s in the "uniform bank" so there isn't really any additional memory to upload into. 

But maybe keeping it as currently written is safer and avoids problems in case of arrray length mismatch (could be driver dependent). So I'm OK with keeping it as is, maybe add a comment to this effect (i.e.. we want the arrays to be of exactly the right length to avoid issues).